### ISSUE TITLE
Why Completeable/Maybe

### ISSUE THREAD

**Issue by ZakTaccardi** [akarnokd/RxJava3-preview#5]
On Android, one of the issues with RxJava is the method count.

Because of this limitation, I don't see the value of `Completable` or `Maybe`.

Both types return once, like `Single<T>`. As a Kotlin developer, representing `T` as a sealed class gives me all the power of a `Completable` or a `Maybe`.

I'd like to see a module with just `Flowable<T>` and `Single<T>`, somehow, if this could drastically reduce method counts.

Just thoughts..

**Comment by akarnokd** [comment]
RxJava 2 proguards well as I heard so method count shouldn't be an issue.

However, if you know any technologies or have the time, we could investigate some library-rewriting options where one can cherry-pick types and methods upfront and depend on a custom-made library.

**Comment by ZakTaccardi** [comment]
> where one can cherry-pick types and methods upfront and depend on a custom-made library

I don't think this would actually be possible in Java, if you wanted to keep the `.toCompletable()` fluent style of calling methods.

Kotlin's extension methods on the other hand..

**Comment by JakeWharton** [comment]
If you never call `.toCompletable()` ProGuard will remove that method and the entire `Completable` type.
